src/MaxText/configs/types.py

```diff
@@ -0,0 +1,1362 @@
+# Copyright 2023–2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pydantic-based configuration system for MaxText, organized into modular classes."""
+
+# pylint: disable=too-many-lines
+
+import os
+import sys
+from enum import Enum
+from tempfile import gettempdir
+from typing import Any, NewType, Literal
+import math
+from math import prod
+
+import jax
+
+from pydantic import (
+    BaseModel,
+    Field,
+    NonNegativeFloat,
+    NonNegativeInt,
+    PositiveInt,
+    model_validator,
+    field_validator,
+)
+from MaxText import accelerator_to_spec_map
+from MaxText.common_types import DecoderBlockType
+
+# ----------------------------------------------------------------------------
+# Reusable Enums and Type Aliases
+# ----------------------------------------------------------------------------
+
+PathStr = str
+AxisNames = NewType("AxisNames", str)
+
+
+class DType(str, Enum):
+  """Supported data types for weights and activations."""
+
+  BFLOAT16 = "bfloat16"
+  FLOAT32 = "float32"
+  FLOAT16 = "float16"
+
+
+class MatmulPrecision(str, Enum):
+  """Precision levels for matrix multiplications."""
+
+  DEFAULT = "default"
+  HIGH = "high"
+  HIGHEST = "highest"
+
+
+class QuantizationType(str, Enum):
+  """Supported quantization schemes."""
+
+  NONE = ""
+  INT8 = "int8"
+  INTMP = "intmp"
+  FP8 = "fp8"
+  NANOO_FP8 = "nanoo_fp8"
+  FP8_NANO_V2 = "fp8_nanoo"
+  FP8_GPU = "fp8_gpu"
+  FP8_FULL = "fp8_full"
+
```

Comment:
Just for my understanding, how did you generate all the options for each type? I see some of these listed in [`base.yml` comments](https://github.com/AI-Hypercomputer/maxtext/blob/9b2f657683f411c260c76dda610715f80376f711/src/MaxText/configs/base.yml#L91-L98), others listed [elsewhere in MaxText](https://github.com/AI-Hypercomputer/maxtext/blob/9b2f657683f411c260c76dda610715f80376f711/docs/explanations/quantization.md?plain=1#L51-L56), etc.

Also, do we have a mismatch in what is [actually allowed to be called](https://github.com/AI-Hypercomputer/maxtext/blob/9b2f657683f411c260c76dda610715f80376f711/src/MaxText/pyconfig.py#L278) in MaxText vs. what is intended to be supported? Or is this method not actually getting picked up?

Also, is